NPE with multiple flavor dimensions

I get this exception with 1.0.0 in a project with multiple flavor dimensions. It seems that the variant which is looked up by name is null.

```
Caused by: java.lang.NullPointerException: Cannot get property 'name' on null object
        at org.robolectric.gradle.RobolectricPlugin$_apply_closure1.doCall(RobolectricPlugin.groovy:29)
```

--- a/src/main/groovy/org/robolectric/gradle/Configuration.groovy
+++ b/src/main/groovy/org/robolectric/gradle/Configuration.groovy
@@ -44,34 +44,12 @@ class Configuration {
     }
 
     /**
-     * Return the all variant names.
+     * Return all variants.
      *
-     * @return  Collection of variant names.
+     * @return  Collection of variants.
      */
-    Collection<String> getVariantNames() {
-        def rval = new ArrayList<String>()
-        for (String type : getBuildTypeNames()) {
-            def flavors = getFlavorNames()
-            if (flavors.isEmpty()) {
-                rval.add("${type}")
-            } else {
-                for (String flavor : flavors) {
-                    rval.add("${flavor}${type.capitalize()}")
-                }
-            }
-        }
-        return rval
-    }
-
-    /**
-     * Return the given variant.
-     *
-     * @param   name    Variant name.
-     * @return  The specified variant.
-     */
-    BaseVariant getVariant(String name) {
-        def variants = hasAppPlugin ? project.android.applicationVariants : project.android.libraryVariants
-        return variants.find { variant -> return variant.name.equals(name) } as BaseVariant
+    Collection<BaseVariant> getVariants() {
+        return hasAppPlugin ? project.android.applicationVariants : project.android.libraryVariants
     }
 
     private Collection<String> getFlavorNames() {

--- a/src/main/groovy/org/robolectric/gradle/RobolectricPlugin.groovy
+++ b/src/main/groovy/org/robolectric/gradle/RobolectricPlugin.groovy
@@ -24,8 +24,7 @@ class RobolectricPlugin implements Plugin<Project> {
             }
 
             // Configure the test tasks
-            for (String name : configuration.getVariantNames()) {
-                def variant = configuration.getVariant(name)
+            configuration.variants.all { variant ->
                 def task = project.tasks.findByName("test${variant.name.capitalize()}")
 
                 // Set RobolectricTestRunner properties

--- a/src/test/groovy/org/robolectric/gradle/RobolectricPluginTest.groovy
+++ b/src/test/groovy/org/robolectric/gradle/RobolectricPluginTest.groovy
@@ -127,6 +127,42 @@ class RobolectricPluginTest {
     }
 
     @Test
+    public void plugin_withFlavorDimensions_configuresTestTasks() {
+        final Project project = createProject()
+        project.android {
+            flavorDimensions "a", "B"
+            productFlavors {
+                a1 {
+                    flavorDimension "a"
+                }
+                A2 {
+                    flavorDimension "a"
+                }
+                B1 {
+                    flavorDimension "B"
+                }
+                b2 {
+                    flavorDimension "B"
+                }
+            }
+        }
+        project.evaluate()
+
+        def test_a1B1_Debug = project.tasks.getByName("testA1B1Debug")
+        assertThat(getPackage(test_a1B1_Debug)).isEqualTo("com.example")
+        assertThat(getAssetPath(test_a1B1_Debug)).endsWith("build/intermediates/assets/a1B1/debug")
+        assertThat(getResourcePath(test_a1B1_Debug)).endsWith("build/intermediates/res/a1B1/debug")
+        assertThat(getManifestPath(test_a1B1_Debug)).endsWith("build/intermediates/manifests/full/a1B1/debug/AndroidManifest.xml")
+
+        def test_A2b2_Release = project.tasks.getByName("testA2B2Release")
+        assertThat(getPackage(test_A2b2_Release)).isEqualTo("com.example")
+        assertThat(getAssetPath(test_A2b2_Release)).endsWith("build/intermediates/assets/A2b2/release")
+        assertThat(getResourcePath(test_A2b2_Release)).endsWith("build/intermediates/res/A2b2/release")
+        assertThat(getManifestPath(test_A2b2_Release)).endsWith("build/intermediates/manifests/full/A2b2/release/AndroidManifest.xml")
+
+    }
+
+    @Test
     public void configuration_supportsSettingAnExcludePattern() {
         final Project project = createProject()
         project.robolectric {
@@ -263,7 +299,7 @@ class RobolectricPluginTest {
     }
 
     @Test
-    public void configuration_supportsIngoreFailures() {
+    public void configuration_supportsIgnoreFailures() {
         final Project project = createProject()
         project.robolectric {
             ignoreFailures true

